stream_buffer.c

```diff
@@ -1000,10 +1006,16 @@ static size_t prvReadMessageFromBuffer( StreamBuffer_t * pxStreamBuffer,
         xNextMessageLength = xBufferLengthBytes;
     }
 
+    /* Use the minimum of the wanted bytes and the available bytes. */
+    xCount = configMIN( xNextMessageLength, xBytesAvailable );
+
     /* Read the actual data. */
-    xReceivedLength = prvReadBytesFromBuffer( pxStreamBuffer, ( uint8_t * ) pvRxData, xNextMessageLength, xBytesAvailable ); /*lint !e9079 Data storage area is implemented as uint8_t array for ease of sizing, indexing and alignment. */
+    xTail = prvReadBytesFromBuffer( pxStreamBuffer, ( uint8_t * ) pvRxData, xCount, xTail); /*lint !e9079 Data storage area is implemented as uint8_t array for ease of sizing, indexing and alignment. */

```

Comment:
I think xTail (and so pxStreamBuffer->xTail) should only be updated here if xCount != 0.  Otherwise the tail is moved even though nothing got read from the buffer - especially when the read is abandoned because the user supplied buffer is too small.

Reply:
Nice catch! I wish MISRA would just let us `return 0` once insufficient space is detected earlier.